siliconcompiler/tools/yosys/syn_asic.py

```diff
@@ -320,14 +310,24 @@ def get_synthesis_corner(chip):
                 corner = chip.get('constraint', 'timing', constraint, 'libcorner',
                                   step=step, index=index)
 
-    if isinstance(corner, (list)):
-        corner = corner[0]
-
     return corner
 
 
-def get_dff_liberty_file(chip):
+def _get_synthesis_library_key(chip, lib, corners):
+    if chip.valid('library', lib, 'option', 'file', 'yosys_synthesis_libraries'):
+        return ('library', lib, 'option', 'file', 'yosys_synthesis_libraries')
 
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+    delaymodel = chip.get('asic', 'delaymodel', step=step, index=index)
+    for corner in corners:
+        if chip.valid('library', lib, 'output', corner, delaymodel):
+            return ('library', lib, 'output', corner, delaymodel)
+
+    return ('library', lib, 'output', corners[0], delaymodel)
```

Comment:
As in `openroad.py` returning invalid library key is fine here?

Reply:
same